### ISSUE TITLE
Don't require template to have variable name

### ISSUE THREAD

**Issue by milotoor** [robrichard/browserify-compile-templates#1]
Hi Rob, thanks for this tool! Very handy!

I'm in the middle of migrating a substantial web application into a Browserify-compatible format, and this transform you've provided is exactly the thing I need. The many-templates-per-file aspect is especially nice, and fits my current template organization perfectly.

There is one suggestion I would like to make. My templates (and many others out in the wild) never use a variable name. They always take the form of

```
var template = _.template("<%= foo %>")
el.html(template({ foo: "bar" }))
```

To my knowledge, there's no way to leverage your compiler and keep the templates in that format. The template would have to change to `<%= obj.foo %>` or something similar.

Have you considered modifying the compile script to not require a variable name? This would help me greatly, as otherwise my only recourse is to update my several hundred templates by prefixing some object name to every variable reference...

Thanks again!

**Comment by robrichard** [comment]
Hi Milo,

This was intentional. The underscore docs warn that using _.template this way will significantly affect performance since it uses the `with` keyword. I don't think the default behavior should allow it. I wouldn't be opposed to a pull request that allows this to work by adding a `data-use-with` attribute to the script tag.

**Comment by milotoor** [comment]
OK, I'll work on that. Thanks Rob!

**Comment by milotoor** [comment]
Rob, what do you think about making `useWith` a configuration parameter? It seems likely to me that anyone looking for this feature would be looking for it to cover all of their templates, and it would save a lot of time and characters not having to add a `data-use-with` attribute to each script tag. Additionally this would allow pre-existing templates to be totally compatible with your transform (my use case), and would help keep the templates decoupled from the transform tool.

Thoughts? I'm happy to put together what I think will be the minimally sufficient changes for this to work.

**Comment by robrichard** [comment]
Sure, sounds good.

**Comment by milotoor** [comment]
See #2. Criticism very welcome!

**Comment by robrichard** [comment]
Thanks! Published 0.2.0

**Comment by milotoor** [comment]
Thank you!
